### ISSUE TITLE
Lectures on Deep Learning

### ISSUE THREAD

**Issue by sarthaksharma13** [iiith-rrc/summer-sessions-2019#8]
Hi, 
Coming Thursday we would start with lectures on Deep Learning. The expectations is that theory uploaded here would be read before coming to the lectures. We would be covering 3 assignments (minimally) during the sessions which would last 5-7 days.
For the Thursday (23 May 2019) session, please read the following before the class :
Intro : http://neuralnetworksanddeeplearning.com/
Chap 1: http://neuralnetworksanddeeplearning.com/chap1.html
Chap 2: http://neuralnetworksanddeeplearning.com/chap2.html

**Comment by talsperre** [comment]
For tomorrow's lecture, bring your own laptops with working internet connection (3G or 4G net). All tutorials will be done on Google Colab so internet connection is a must.

**Comment by sarthaksharma13** [comment]
Here are a few reads to be done for different optimization methods :
1. https://towardsdatascience.com/neural-network-optimization-algorithms-1a44c282f61d
2. http://fa.bianp.net/teaching/2018/eecs227at/

**Comment by sarthaksharma13** [comment]
Code for today's class (27/5/19) :
1. CIFAR10 : https://colab.research.google.com/drive/18dgQudsJr3t0NPdzaW0CUyLzDoTlc4ls
2. Finetune CIFAR 100 : https://colab.research.google.com/drive/10hmDESmHoxDl3bCnfjFTHHr766pbgeu8

**Comment by sarthaksharma13** [comment]
Task 1 : 
Aim : To get familiar with different network designs and hyperparameter search. Try methods to normalize the data.
Have a look at different architectures like VGGNet, AlexNet, ResNet etc.
- Try min 2 network definitions with min 5 hyperparameters each. Try to push the test accuracy above 80%. 
  Do this for both CIFAR10 and CIFAR100.


Task 2:
Aim : To get familiar with performance of the network when variation in the dataset occurs. Try to reason about that using plots.
 - Train a network for the standar MNIST dataset. Test it on the test split.
 - Test the same network on the rotated MNIST dataset (w/o training on it) Report the performance. 
 - After this, report graph(s) that adds a certain percentage(0%,10%,20%..) of rotated dataset to the original dataset (X axis) and 
   report performance on a test split consisting of only rotated, normal and a mixure of both normal and rotated dataset(Y axis).


Deadline for both the tasks : 3 June 11:59 PM.

**Comment by talsperre** [comment]
Read this lesson once before coming for tomorrow's lecture: 
1. https://colah.github.io/posts/2015-08-Understanding-LSTMs/

**Comment by talsperre** [comment]
Bring your own laptops with working internet connection (3G or 4G net). All tutorials will be done on Google Colab so internet connection is a must.

**Comment by talsperre** [comment]
The notebook for todays class: 
1. https://colab.research.google.com/drive/1HHAM10nxpxA4u0uKrVGZJl9EEPWpRydv

**Comment by AnirudhaRamesh** [comment]
Could we be given access to ADA anytime soon? Training our models for this assignment and beyond is taking very long with our local machines + colab, floydhub etc free tier platforms. 
Thanks in advance

**Comment by talsperre** [comment]
Send a mail to hpc@iiit.ac.in in the following format.
```
Please make an ADA account for me as I need it for pursuing my research. The details are 
given below:

Name: X
Research Center: RRC (Robotics Research Center)
Faculty Advisor: Professor Madhava Krishna
Preferred Login Id: X
```
